lib/containers/utils.pm

```diff
@@ -79,6 +79,50 @@ sub registry_url {
     return sprintf("%s/%s:%s", $repo, $container_name, $version_tag);
 }
 
+sub test_update_cmd {
+    my %args = @_;
+    my $runtime = $args{runtime};
+    my $container = $args{container};
+
+    # podman update was added to v4.3.0
+    if ($runtime eq 'podman') {
+        my $version = script_output "podman version | awk '/^Version:/ { print \$2 }'";
+        if (package_version_cmp($version, '4.3.0') <= 0) {
+            record_info("The update command is not supported on podman $version");
+            return;
+        }
+    } elsif ($runtime ne 'docker') {
+        die "Unknown $runtime";
+    }
+
+    # The default values for these options are 0.
+    # Before adding more options check that they're supported by both podman & docker
+    my @opts = ('cpu-shares', 'memory-swap');
+
+    foreach my $opt (@opts) {
+        # Transform 'cpu_shares' into 'CpuShares'
+        (my $param = $opt) =~ s/^(.)(.*)-(.)(.*)/\u$1$2\u$3$4/;
+
+        my $old_value = script_output "$runtime container inspect -f '{{.HostConfig.$param}}' $container";
+        die "Default for $opt != 0" if ($old_value != 0);
+
+        # 10 is the minimum value for blkio-weight
+        my $try_value = ($opt eq 'memory-swap') ? -1 : 10;
```

Comment:
> I don't understand the default value -1 in case not memory-swap. The code also checks cpu-shares

-1 is for memory-swap and 10 is for everything else.